// @s-ui/react-atom-progress-bar/lib/index.js
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/objectWithoutPropertiesLoose";
import React from 'react';
import PropTypes from 'prop-types';
import LineProgressBar from './LineProgressBar';
import CircleProgressBar from './CircleProgressBar';
var TYPES = {
  CIRCLE: 'circle',
  LINE: 'line'
};
var SIZES = {
  LARGE: 'large',
  SMALL: 'small'
};

var AtomProgressBar = function AtomProgressBar(_ref) {
  var type = _ref.type,
      size = _ref.size,
      props = _objectWithoutPropertiesLoose(_ref, ["type", "size"]);

  switch (type) {
    case TYPES.LINE:
      return React.createElement(LineProgressBar, props);

    case TYPES.CIRCLE:
      return React.createElement(CircleProgressBar, Object.assign({
        size: size
      }, props));

    default:
      return React.createElement(LineProgressBar, props);
  }
};

AtomProgressBar.displayName = 'AtomProgressBar';
AtomProgressBar.defaultProps = {
  size: SIZES.LARGE,
  type: TYPES.LINE
};
export default AtomProgressBar;
export { TYPES as atomProgressBarTypes, SIZES as atomProgressBarSizes };
